src/symbolize/gimli/elf.rs

```diff
@@ -421,3 +449,53 @@ fn locate_debugaltlink(path: &Path, filename: &[u8], build_id: &[u8]) -> Option<
 
     locate_build_id(build_id)
 }
+
+fn convert_path<R: gimli::Reader>(r: &R) -> Result<PathBuf, gimli::Error> {
+    let bytes = r.to_slice()?;
+    Ok(PathBuf::from(OsStr::from_bytes(&bytes)))
+}
+
+pub(super) fn handle_split_dwarf<'data>(
+    package: Option<&gimli::DwarfPackage<EndianSlice<'data, Endian>>>,
+    stash: &'data Stash,
+    load: addr2line::SplitDwarfLoad<EndianSlice<'data, Endian>>,
+) -> Option<Arc<gimli::Dwarf<EndianSlice<'data, Endian>>>> {
+    if let Some(dwp) = package.as_ref() {
+        if let Ok(Some(cu)) = dwp.find_cu(load.dwo_id, &load.parent) {
+            return Some(Arc::new(cu));
+        }
+    }
+
+    let mut path = PathBuf::new();
+    if let Some(p) = load.comp_dir.as_ref() {
+        if let Ok(p) = convert_path(p) {
+            path.push(p);
+        }
+    }
+
+    if let Some(p) = load.path.as_ref() {
+        if let Ok(p) = convert_path(p) {
+            path.push(p);
+        }
+    }
```

Comment:
If either `comp_dir` or `path` are `None`, or if `convert_path` fails, then we should immediately return `None`.

Reply:
perhaps for `path=None` and `convert_path` failing, but I think `comp_dir=None` is legal.

Reply:
I think you're right.

Reply:
Alright, addressed.